x/incentives/keeper/gauge.go

```diff
@@ -127,9 +127,42 @@ func (k Keeper) CreateGauge(ctx sdk.Context, isPerpetual bool, owner sdk.AccAddr
 		return 0, types.ErrZeroNumEpochsPaidOver
 	}
 
+	// If the gauge has no lock, then we assume it is a concentrated pool and ensure
+	// the gauge "lock" duration is an authorized uptime.
+	isConcentratedPoolGauge := distrTo.LockQueryType == lockuptypes.NoLock
+
+	// If the gauge has an internal gauge denom, it is an internal gauge
+	// and should be run through different validation logic (see below).
+	//
+	// Two important reminders/assumptions:
+	// 1. `NoLock` gauges are required to have empty denoms in `ValidateBasic`, so this
+	// check cannot be controlled by user input.
+	// 2. The safety of this leans on the special-casing of internal gauge logic during
+	// distributions, which should be using the internal incentive duration gov param instead of the duration value.
+	isInternalConcentratedPoolGauge := distrTo.Denom == types.NoLockInternalGaugeDenom(poolId)
+	isExternalConcentratedPoolGauge := isConcentratedPoolGauge && !isInternalConcentratedPoolGauge
+
 	// Ensure that this gauge's duration is one of the allowed durations on chain
-	durations := k.GetLockableDurations(ctx)
-	if distrTo.LockQueryType == lockuptypes.ByDuration {
+	// Concentrated pool gauges check against authorized uptimes (if external) or
+	// epoch duration (if internal).
+	//
+	// All other gauges check against the default set of lockable durations.
+	var durations []time.Duration
+	if isExternalConcentratedPoolGauge {
+		durations = k.clk.GetParams(ctx).AuthorizedUptimes
+	} else if isInternalConcentratedPoolGauge {
+		// Internal CL gauges use epoch time as their duration. This is a legacy
+		// property that does not affect the uptime on created records, which is
+		// determined by the gov param for internal incentive uptimes.
+		durations = []time.Duration{k.GetEpochInfo(ctx).Duration}
+	} else {
+		// This branch is applicable to CFMM pool types such as balancer and stableswap.
+		durations = k.GetLockableDurations(ctx)
+	}
```

Comment:
This would make the conditional branching stricter and eliminate the likelihood of invalid field setup causing issues.

nit: the `else` branch is also not covered by a test at the moment and might benefit from it

```suggestion
	if isExternalConcentratedPoolGauge {
		durations = k.clk.GetParams(ctx).AuthorizedUptimes
	} else if isInternalConcentratedPoolGauge {
		// Internal CL gauges use epoch time as their duration. This is a legacy
		// property that does not affect the uptime on created records, which is
		// determined by the gov param for internal incentive uptimes.
		durations = []time.Duration{k.GetEpochInfo(ctx).Duration}
	} else if !isConcentratedPoolGauge {
		// This branch is applicable to CFMM pool types such as balancer and stableswap.
		durations = k.GetLockableDurations(ctx)
	} else {
		return 0, fmt.Errorf("invalid gauge type: %s", distrTo.LockQueryType)
	}
```

Reply:
I see what you're getting at here, but this flow is not intended to serve as a check or filter where we return an error. For example, your implementation would error if either `CreateGauge` is called for a Group gauge, which is not what we want here.

It does seem as though this logic is somewhat unclear though. I moved this check into the block where the duration checks happen, as that probably makes it clearer (i.e. `if NoLock gauge || ByDuration gauge, run this logic` implies that we are covering everything we need to with the logic branches in the original implementation). Hopefully this makes it a bit clearer.

Please let me know if I misunderstood your suggestion

Reply:
34bba93dc0d82eb528c4a9c123a27abc6d912dbb